docs/glossary.md

```diff
@@ -0,0 +1,100 @@
+
+A liturgical day generally aligns with a day in the Gregorian calendar but has some nuances: it typically begins at First Vespers or Vigils (if observed), or otherwise at midnight, and ends at the beginning of the following liturgical day. If a liturgical day begins before midnight, it remains identified as a single liturgical day, even if it spans two civil days. A liturgical day is identified by a Gregorian date (considering the majority of the liturgical day that occurs within a given civil day).
+
+In Romcal, it is this concept of the liturgical day that is calculated within a liturgical calendar.
+
+> [!NOTE]
+> A liturgical day should not be confused with a celebration (see [celebration](#celebration)).
+
+## Liturgical Period
+
+In Romcal, in addition to the liturgical seasons, liturgical periods provide further detail and metadata about the current liturgical period. For example, the octaves of Christmas and Easter, the period before and after Epiphany, Holy Week, etc. They also denote periods traditionally observed in the Catholic Church and by religious communities (especially for Marian hymns), including the period from the Baptism of Christ to the Presentation of Jesus in the Temple, and from the Presentation of Jesus in the Temple to Holy Thursday.
+
+## Precedence
+
+The principle of precedence is used to determine which liturgical celebration should be observed when two or more celebrations fall on the same day. The principle of precedence is based on the liturgical rank of the celebrations.
+
+See [GNLYC §59](https://www.liturgyoffice.org.uk/Resources/GIRM/Documents/GNLY.pdf#page=14).
+
+## Proper of Saints
+
+The Proper of Saints (or Sanctoral) consists of the fixed feasts, celebrated on the very same date each year (no matter what the day of the week),including Christmas and all the saints’ days.
+
+## Proper of Time
+
+The Proper of Time (or Temporal) consists of the movable feasts, most of them keyed to Easter (which falls on a different Sunday every year), including Ascension, Pentecost, and so on.
+
+> [!NOTE]
+> We avoid using the term Temporal to prevent potential confusion with the `Temporal` object in ECMAScript, which is still in a [proposal stage](https://tc39.es/proposal-temporal/docs/).
+
+## Psalter Week Cycle
+
+The four-week cycle of the psalter is coordinated with the liturgical year in such a way that on the First Sunday of Advent, the First Sunday in Ordinary Time, the First Sunday of Lent, and Easter Sunday the cycle is always begun again with Week 1 (others being omitted when necessary).
+
+See [GILH §133](https://www.liturgyoffice.org.uk/Resources/Rites/GILH.pdf#page=26).
+
+## Liturgical Rank
```

Comment:
With some perspective now, I prefer that we use short names and not necessarily prefix with the word `Liturgy`. Since this project is already related to liturgy, it would be redundant. And it's also better to have short names in the codebase.

Unless it could cause confusion (for example, it's better to keep `LiturgicalDay`, because `Day` alone indicates a civil day).

So here, I would prefer to revert to `Rank` rather than `LiturgicalRank`.

Reply:
working through this script you implemented @tukusejssirs (which has been added on this PR). it doesn't handle h1's, so the top-level header of "Glossary" is removed upon sort.... just letting you know, i'm working on a fix.

Reply:
i've got it sorted now, i've also pushed the code for warning you that it's not sorted when you try to commit